lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java

```diff
@@ -74,18 +75,49 @@
     throw new IllegalStateException("Unable to parse composition", throwable);
   };
 
-  private final LottieListener<LottieComposition> loadedListener = this::setComposition;
+  private final LottieListener<LottieComposition> loadedListener = new WeakSuccessListener(this);
 
-  private final LottieListener<Throwable> wrappedFailureListener = new LottieListener<Throwable>() {
-    @Override
-    public void onResult(Throwable result) {
-      if (fallbackResource != 0) {
-        setImageResource(fallbackResource);
+  private static class WeakSuccessListener implements LottieListener<LottieComposition> {
+
+    private final WeakReference<LottieAnimationView> targetReference;
+
+    public WeakSuccessListener(LottieAnimationView target) {
+      this.targetReference = new WeakReference<>(target);
+    }
+
+    @Override public void onResult(LottieComposition result) {
+      LottieAnimationView targetView = targetReference.get();
+      if (targetView == null) {
+        return;
       }
-      LottieListener<Throwable> l = failureListener == null ? DEFAULT_FAILURE_LISTENER : failureListener;
+      targetView.setComposition(result);
+    }
+  }
+
+  private final LottieListener<Throwable> wrappedFailureListener = new WeakFailureListener(this);
```

Comment:
>I think you can omit it since there isn't a non-wrapped version that people could confuse this with but other than that

I wanted to apply your suggestion, but realized there is actually a non-wrapped version called `failureListener` which comes from the user. It seems like the non-wrapped is the one passed by the user, and the wrapped is also responsible for setting a fallback if resource fails to load. 
Looking at public api of `LottieAnimationView` - it's possible to set multiple `LottieOnCompositionLoadedListener` callbacks, but only one `setFailureListener`, which explain why wrapped variant is needed here.

Reply:
Thanks! Haven't looked at this code in a while 😄